plugins/setting-resources/src/components/Backup.svelte

```diff
@@ -0,0 +1,355 @@
+<!--
+// Copyright © 2025 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+-->
+<script lang="ts">
+  import { AccountRole, getCurrentAccount, hasAccountRole } from '@hcengineering/core'
+  import { getMetadata } from '@hcengineering/platform'
+  import presentation from '@hcengineering/presentation'
+  import { Breadcrumb, Button, Expandable, Header, Label, Link, Loading } from '@hcengineering/ui'
+  import Scroller from '@hcengineering/ui/src/components/Scroller.svelte'
+  import view from '@hcengineering/view'
+  import { onMount } from 'svelte'
+  import setting from '../plugin'
+  import { BackupInfo, BackupSnapshot } from '../types'
+
+  let loading = true
+
+  const backupUrl = getMetadata(setting.metadata.BackupUrl) ?? ''
+  const token = getMetadata(presentation.metadata.Token) ?? ''
+  const owner = hasAccountRole(getCurrentAccount(), AccountRole.Owner)
+  const workspaceId = getMetadata(presentation.metadata.WorkspaceId) ?? ''
+
+  let backupInfo:
+  | {
+    files: { name: string, size: number }[]
+    error?: string
+    info?: BackupInfo
+  }
+  | undefined
+
+  $: fileSizes = new Map(backupInfo?.files?.map((it) => [it.name, it.size ?? 0]))
+
+  function formatDate (timestamp: number): string {
+    return new Date(timestamp).toLocaleString()
+  }
+
+  function getSize (size: number): string {
```

Comment:
We have an utility for this somewhere, used in Drive

Reply:
Let it be for now.